src/bitcoin-tx.cpp

```diff
@@ -287,7 +389,26 @@ static void MutateTxAddOutScript(CMutableTransaction& tx, const string& strInput
 
     // extract and validate script
     string strScript = strInput.substr(pos + 1, string::npos);
-    CScript scriptPubKey = ParseScript(strScript); // throws on err
+    CScript scriptPubKey; // The scriptPubKey in the txOut
+    CScript script = ParseScript(strScript);
+    // This could either refer to the scriptPubKey (for a bare pay-to-script)
+    // or the redeem script (for a P2SH or P2WSH)
+
+    if (paytoscripthash) {
```

Comment:
as discussed, this function `MutateTxAddOutScript()` can be used to add the following kinds of TXO:

- bare script
- P2SH
- P2WSH

Currently, bitcoin-tx can only create bare script outputs. This flag `paytoscripthash` is used to signal that the TXO should be P2SH rather than a pay to script.